experimental/bot/logger/config.go

```diff
@@ -3,19 +3,25 @@
 
 package logger
 
+// Config contains all configuration needed by the logger.
 type Config struct {
 	// AdminLogLevel is "debug", "info", "warn", or "error".
 	AdminLogLevel string
 
 	// AdminLogVerbose: set to include full context with admin log messages.
 	AdminLogVerbose bool
+
+	// TelemetryLogLevel is "debug", "info", "warn", or "error".
+	TelemetryLogLevel string
```

Comment:
1/5 prefer an wrapper `NewTelemetryLogger(Logger)Logger` to options and bundling everything in 1 type. TelemtryLogger could be completely separated, no?

Reply:
On that note, the `MessageToAdmin` business could be implemented as a `CcLogger` with its own way of initialization (a list of teams - channels/users)

Reply:
If I understand your proposal correctly, the code would now works like this

```go
log := logger.New(api, logger.LogToTelemetry("debug", tracker))
```
vs. 
```go
log := logger.NewTelemetryLogger("debug", tracker, logger.New(api))
```
is what you are proposing.

I like the functions options more, as it makes clear that a logger can be an admin and telemetry logger.

Reply:
If I understand correctly, the end result would be something like:
```
func NewTelemetryLogger(logger Logger, logLevel string, tracker telemetry.Tracker) Logger {
   return logger.New(logger, LogToTelemetry(logLevel, tracker)
}
```

The problem with that is that Logger is a interface, so I cannot affect the inners of the object. I could go with a sealed interface, but would make the code much more complicated. The other option I see is instead of "promoting a logger" having a utility function to make faster what you want.
```
func NewTelemetryLogger(api common.LogAPI, logLevel string, tracker telemetry.Tracker) Logger {
   return logger.New(api, LogToTelemetry(logLevel, tracker))
}
```

Reply:
There is another option to "promote the logger", adding a WithTracker or WithAdminCC function to the Interface. That way we could do either:
```
l := New(client)
l.WithTracker(LogLevelDebug, tracker)
```
or
```
l := New(client)
l = NewTelemetryLogger(l, LogLevelDebug, tracker)
```
and NewTelemetryLogger would be defined as:
```
func NewTelemetryLogger(logger Logger, logLevel string, tracker telemetry.Tracker) Logger {
   return logger.WithTracker(logLevel, tracker)
}
```

Reply:
After discussing offline with @levb , I have made some changes to decouple the default logger from the telemetry and admincc loggers. @hanzei , I re-request your review since the approach is quite different, in case you want to give more comments.